bokeh/models/ranges.py

```diff
@@ -119,6 +142,16 @@ class FactorRange(Range):
     this categorical range.
     """)
 
+    bound_lower = Either(String, Int, help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
+
+    bound_upper = Either(String, Int, help="""
+    The max value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
```

Comment:
> except that now users are always required to supply both bounds what if they only want a bound below

Good point, my vote for that would be to keep to the `None` convention meaning no bound - so `bounds=(None, 10)` in this case its imperative that first is min, second is max so we should definitely use that convention as you suggest.

If we use `None` as I suggest above, spelling all that out will be:
`Either(Tuple(Either(Float, None), Either(Float, None)), Tuple(Either(Datetime, None), Either(Datetime, None)), Tuple(Either(Int, None), Either(Int, None)), Auto, None, default=Auto)`
which is definitely a lot so I guess we'll definitely need a new property or model for `Bounds` as you suggest, welcome a nudge on which you think is best.